components/kyma-environment-broker/internal/environmentscleanup/broker/client.go

```diff
@@ -0,0 +1,125 @@
+package broker
```

Comment:
Did you consider just reuse `/internal/broker/instance_deprovision.go` instead call to the broker? 
You can simply extract the common part of the logic and create two "connections" to it, one for HTTP queries (broker) and one directly from the code